### DISCUSSION TITLE
部署了单节点的dolphinscheduler，版本3.1.6，在配置数据源的时候测试成功，使用时却没有密码，请问是怎么回事

### DISCUSSION THREAD

**Discussion by leerelisten** [apache/dolphinscheduler#14151] [Q&A]
1.配置的数据源如图所示，测试成功，进行保存
![image](https://github.com/apache/dolphinscheduler/assets/43003174/2fcb6d24-83ea-48b5-a8fa-74852198c3f1)
2.查看后台元数据表，发现数据已经保存上了
![image](https://github.com/apache/dolphinscheduler/assets/43003174/785822b0-c6ba-4ee5-8c0b-32e47da23be1)

3.再次打开，或者执行任务时，发现连接不上，查看密码不存在。
![image](https://github.com/apache/dolphinscheduler/assets/43003174/6057df4b-5012-4149-9436-b7d059bf456f)
![image](https://github.com/apache/dolphinscheduler/assets/43003174/15d87d3e-c316-48da-bfd8-6b61704d3a1d)
4.请看一下是什么问题，如何解决

**Comment by zhongjiajie** [comment]
when you edit the data source, the expected behavior is without a password, which means you have to add a password again when you edit the data source to make sure you are own the data source

And I see that when you run the job, the password already fill in, could you make sure your worker network can connection the datasource?

**Reply by leerelisten** [reply]
i am sure my working network can connect the datasource . In mysql database ,i see the password saved , but when i test the connection it does't work and i can not connect to the datasource . the scheduler logs said that "USING PASSWORD :NO".

**Reply by F7kyyy** [reply]
解决了么，我在3.2.0好像也有这个问题
